### PR TITLE
Expanding documentation in queue.go

### PR DESCRIPTION
**Pull request by cassiozareck**
A set of terminology, along with a broader description, can help more people engage with the Gitea queue system, providing insights and ensuring its correct use.

### CODE REVIEW

File: modules/queue/queue.go [hunk 1]
```diff
@@ -1,27 +1,62 @@
...
 //
-// There are two major kinds of concepts:
+// Terminology:
 //
-// * The "base queue": channel, level, redis:
-//   - They have the same abstraction, the same interface, and they are tested by the same testing code.
-//   - The dummy(immediate) queue is special, it's not a real queue, it's only used as a no-op queue or a testing queue.
+//  1. Task:
+//     -	A task can be a simple value, such as an integer, or a more complex structure that has multiple fields and
+//     methods. The aim of a task is to be a unit of work, a set of tasks will be sent to a handler to be processed.
```
Comment (wxiaoguang) [inline modules/queue/queue.go]: IIRC there is no "task" concept in the queue system. Queue system only handles "item", which is a binary slice in the queue

File: modules/queue/queue.go [hunk 2]
```diff
@@ -1,27 +1,62 @@
...
-//   - It calls the "handler" to process the data in the base queue.
-//   - Its "Push" function doesn't block forever,
-//     it will return an error if the queue is full after the timeout.
+//  2. Batch:
+//     - A collection of tasks that are grouped together for processing. Each worker receives a batch of tasks.
+//
+//  3. Worker:
+//     - Individual unit of execution designed to process tasks from the queue. It's a goroutine that calls the Handler
+//     - Workers will get new tasks through a channel (WorkerPoolQueue is responsible for the distribution)
+//     - As workers operate in parallel, the default value of max workers is n/2, where n is the number of logical CPUs
```
Comment (wxiaoguang) [inline modules/queue/queue.go]: This assumption is not done by queue system, but by the setting system.

File: modules/queue/queue.go [hunk 3]
```diff
@@ -1,27 +1,62 @@
...
+//     - When processing these batches, there might be instances where certain tasks remain unprocessed or "unhandled".
+//     In such scenarios, the Handler ensures these unhandled tasks are returned to the base queue after a brief delay.
+//     This mechanism is particularly beneficial in cases where the processing entity (like a document indexer) is
+//     temporarily unavailable. It ensures that no task is skipped or lost due to transient failures in the processing
+//     mechanism.
+//
+//  5. Base queue:
+//     - Represents the underlying storage mechanism for the queue. There are several implementations:
+//     - Channel: Uses Go's native channel constructs to manage the queue, suitable for in-memory queuing.
+//     - Level, Redis: Especially useful in persistent queues and clusters, where we may have multiple nodes.
```
Comment (wxiaoguang) [inline modules/queue/queue.go]: Levelqueue/leveldb is a disk-only storage, it only works for one instance.

File: modules/queue/queue.go [hunk 4]
```diff
@@ -1,27 +1,62 @@
...
+//     - Dummy: This is special, it's not a real queue, it's only used as a no-op queue or a testing queue.
```
Comment (wxiaoguang) [inline modules/queue/queue.go]: Or it is used as an "immediate" queue

**Review by wxiaoguang — APPROVED** [review]

**Review by lunny — APPROVED** [review]

**Pull request merged**